src/utils/thread_helper.cpp

```diff
@@ -202,10 +226,10 @@ ThreadHelper::ThreadHelper(otInstance *aInstance, otbr::Ncp::ControllerOpenThrea
     : mInstance(aInstance)
     , mNcp(aNcp)
 {
-#if OTBR_ENABLE_TELEMETRY_DATA_API && OTBR_ENABLE_NAT64
+#if OTBR_ENABLE_TELEMETRY_DATA_API && OTBR_ENABLE_NAT64 || OTBR_ENABLE_DHCP6_PD
```

Comment:
Do you want to wrap the condition like (OTBR_ENABLE_NAT64 || OTBR_ENABLE_DHCP6_PD)?